waspc/test/Generator/PackageJsonGeneratorTest.hs

```diff
@@ -2,35 +2,70 @@ module Generator.PackageJsonGeneratorTest where
 
 import Test.Tasty.Hspec
 import qualified Wasp.AppSpec.App.Dependency as D
-import Wasp.Generator.PackageJsonGenerator (resolveNpmDeps)
+import Wasp.Generator.PackageJsonGenerator
+  ( DependencyConflictError (DependencyConflictError),
+    resolveDependencies,
+    resolveNpmDeps,
+  )
 
 spec_resolveNpmDeps :: Spec
 spec_resolveNpmDeps = do
   let waspDeps =
-        [ ("axios", "^0.20.0"),
-          ("lodash", "^4.17.15")
+        D.fromList
+          [ ("a", "1"),
+            ("b", "2")
+          ]
+
+  it "a conflicting version number is detected" $ do
+    let userDeps =
+          D.fromList
+            [ ("a", "1"),
+              ("b", "3")
+            ]
+    resolveDependencies waspDeps userDeps
+      `shouldBe` Left
+        [ DependencyConflictError
+            (D.create ("b", "2"))
+            (D.create ("b", "3"))
         ]
+  it "same versions so no conflict" $ do
+    let userDeps =
+          D.fromList
+            [ ("a", "1"),
+              ("b", "2")
+            ]
+    resolveDependencies waspDeps userDeps
+      `shouldBe` Right (waspDeps, [])
 
-  it "Concatenates two distincts lists of deps." $ do
+  it "no version overlap so no conflict" $ do
     let userDeps =
-          [ ("foo", "bar"),
-            ("foo2", "bar2")
-          ]
-    resolveNpmDeps (D.fromList waspDeps) (D.fromList userDeps)
-      `shouldBe` Right (D.fromList waspDeps, D.fromList userDeps)
+          D.fromList
+            [ ("c", "1"),
+              ("d", "2")
+            ]
+    resolveDependencies waspDeps userDeps
+      `shouldBe` Right (waspDeps, userDeps)
 
-  it "Does not repeat dep if it is both user and wasp dep." $ do
+  it "partial overlap without conflict" $ do
```

Comment:
What I liked better about the old name of this test is that it captured what property is being tested: "Does not repeat dep if it is both user and wasp dep". With this new name I know which situation it is testing, but I don't know what is the actual property being tested.